File: src/main/java/racingcar/model/parser/CarsParser.java
package racingcar.model.parser;

import java.util.Arrays;
import java.util.List;
import racingcar.model.car.Car;

public class CarsParser {

    public static List<Car> parse(final String input) {
        final String[] carNames = input.split(",");
        return Arrays.stream(carNames)
                .map(Car::new)
                .toList();
    }
}

File: src/main/java/racingcar/model/parser/CarsParser.kt
package racingcar.model.parser

import racingcar.model.car.Car
import java.util.*

object CarsParser {
    private const val DELIMITER: String = ","

    @JvmStatic
    fun parse(input: String): List<Car> {
        val carNames = input.split(DELIMITER.toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        return Arrays.stream(carNames)
            .map { name: String? -> Car(name!!) }
            .toList()
    }
}
